-1

**Failed Tests**:   Build  ClangBuild 
**Summary**: https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-3a8707/13163/summary.html
**COMMIT**: e87b00f95d751c6d24f05468e703d126b6d800d0
**CMSSW**: CMSSW_11_1_X_2021-02-28-0000/slc7_amd64_gcc820
**User test area**: For local testing, you can use `/cvmfs/cms-ci.cern.ch/week0/cms-sw/cmssw/33022/13163/install.sh` to create a dev area with all the needed externals and cmssw changes.


## Build

I found compilation error when building: 

<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2021-02-28-0000/src/L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2021-02-28-0000/src/L1Trigger/L1CaloTrigger/plugins/L1TS2PFJetInputPatternWriter.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2021-02-28-0000/src/L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetCalibrator.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2021-02-28-0000/src/L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetProducer.cc 
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2021-02-28-0000/src/L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc: In member function 'bool L1EGCrystalClusterEmulatorProducer::passes_iso(float, float)':
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2021-02-28-0000/src/L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc:1188:5: error: 'is_iso' was not declared in this scope
     is_iso = true;
     ^~~~~~
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2021-02-28-0000/src/L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc:1188:5: note: suggested alternative: 'isinf'
     is_iso = true;
     ^~~~~~
</pre>

## Clang Build

I found compilation error while trying to compile with clang. Command used:
```
USER_CUDA_FLAGS='--expt-relaxed-constexpr' USER_CXXFLAGS='-Wno-register -fsyntax-only' scram build -k -j 32 COMPILER='llvm compile'

```


<pre>>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2021-02-28-0000/src/L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2021-02-28-0000/src/L1Trigger/L1CaloTrigger/plugins/L1EGammaEEProducer.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2021-02-28-0000/src/L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetCalibrator.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2021-02-28-0000/src/L1Trigger/L1CaloTrigger/plugins/L1TS2PFJetInputPatternWriter.cc 
>> Compiling edm plugin /data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2021-02-28-0000/src/L1Trigger/L1CaloTrigger/plugins/Phase1L1TJetProducer.cc 
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2021-02-28-0000/src/L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc:1188:5: error: use of undeclared identifier 'is_iso'
    is_iso = true;
    ^
/data/cmsbld/jenkins/workspace/ib-run-pr-tests/CMSSW_11_1_X_2021-02-28-0000/src/L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc:1189:10: error: use of undeclared identifier 'is_iso'
  return is_iso;
         ^
</pre>

--- a/L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc
+++ b/L1Trigger/L1CaloTrigger/plugins/L1EGammaCrystalsEmulatorProducer.cc
@@ -1177,12 +1177,13 @@ void L1EGCrystalClusterEmulatorProducer::produce(edm::Event& iEvent, const edm::
 }
 
 bool L1EGCrystalClusterEmulatorProducer::passes_iso(float pt, float iso) {
+  bool is_iso = true;
   if (pt < slideIsoPtThreshold) {
     if (!((a0_80 - a1_80 * pt) > iso))
-      return false;
+      is_iso = false;
   } else {
     if (iso > a0)
-      return false;
+      is_iso = false;
   }
   if (pt > plateau_ss)
     is_iso = true;

